base/applications/mspaint/dib.cpp

```diff
@@ -333,3 +333,61 @@ HBITMAP SkewDIB(HDC hDC1, HBITMAP hbm, INT nDegree, BOOL bVertical)
     DeleteDC(hDC2);
     return hbmNew;
 }
+
+typedef struct tagBITMAPINFODX
+{
+    BITMAPINFOHEADER bmiHeader;
+    RGBQUAD          bmiColors[256];
+} BITMAPINFODX, *LPBITMAPINFODX;
+
+HGLOBAL BitmapToClipboardDIB(HBITMAP hBitmap)
+{
+    BITMAP bm;
+    if (!GetObject(hBitmap, sizeof(BITMAP), &bm))
+        return NULL;
+
+    BITMAPINFODX bmi;
+    ZeroMemory(&bmi, sizeof(bmi));
+    bmi.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bmi.bmiHeader.biWidth = bm.bmWidth;
+    bmi.bmiHeader.biHeight = bm.bmHeight;
+    bmi.bmiHeader.biPlanes = 1;
+    bmi.bmiHeader.biBitCount = bm.bmBitsPixel;
+    bmi.bmiHeader.biCompression = BI_RGB;
+    bmi.bmiHeader.biSizeImage = bm.bmWidthBytes * bm.bmHeight;
+
+    INT cColors;
+    if (bm.bmBitsPixel < 16)
+        cColors = 1 << bm.bmBitsPixel;
+    else
+        cColors = 0;
+
+    DWORD cbColors = cColors * sizeof(RGBQUAD);
+    DWORD dwSize = sizeof(BITMAPINFOHEADER) + cbColors + bmi.bmiHeader.biSizeImage;
+    HGLOBAL hGlobal = GlobalAlloc(GHND | GMEM_SHARE, dwSize);
+    if (hGlobal)
+    {
+        LPBYTE pb = (LPBYTE)GlobalLock(hGlobal);
+        if (pb)
+        {
+            CopyMemory(pb, &bmi, sizeof(BITMAPINFOHEADER));
+            pb += sizeof(BITMAPINFOHEADER);
+
+            CopyMemory(pb, bmi.bmiColors, cbColors);
```

Comment:
I wonder where the contents of the bmiColors array gets initialized.

Reply:
My mistake. I will fix tommorow.

Reply:
Fixed.

Reply:
Fixed.